CHANGELOG.md

```diff
@@ -18,6 +18,47 @@ Increment the:
 * [SDK] Add tracer scope configurator
   [#3137](https://github.com/open-telemetry/opentelemetry-cpp/pull/3137)
 
+* [SDK] Support OTEL_SDK_DISABLED environment variable
+  [#3245](https://github.com/open-telemetry/opentelemetry-cpp/pull/3245)
+
+Important changes:
+
+* [SDK] Support OTEL_SDK_DISABLED environment variable
+  [#3245](https://github.com/open-telemetry/opentelemetry-cpp/pull/3245)
+
+  * The SDK now exposes the following new methods:
+
+    * opentelemetry::sdk::trace::Provider::SetTracerProvider()
+    * opentelemetry::sdk::metrics::Provider::SetMeterProvider()
+    * opentelemetry::sdk::logs::Provider::SetLoggerProvider()
+
+  * These methods do support the `OTEL_SDK_DISABLED` environment variable,
+    unlike the corresponding existing API Provider classes.
+
+  * Applications are encouraged to migrate from the API to the SDK
+    `Provider` classes, to benefit from this feature.
+
+  * Note that the API methods will no longer be public starting
+    with OPENTELEMETRY_ABI_VERSION >= 2, only the SDK methods will
```

Comment:
I think we shouldn't remove the API method. As the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#tracerprovider) requires this. It should be instead ok to recommend user to use the SDK provided methods if they need support to enable/disable SDK.

``
Normally, the TracerProvider is expected to be accessed from a central place. Thus, the API SHOULD provide a way to set/register and access a global default TracerProvider.
``

Reply:
Thanks for the review.

For this PR, I reverted the changes that removed the Setter from the API, to avoid blocking the OTEL_SDK_DISABLED change itself.

The spec needs revisions in my opinion, so I filed an issue for this:

* https://github.com/open-telemetry/opentelemetry-specification/issues/4400

Once the spec discussion is settled, we can revisit this part.